### PR TITLE
Refactored editor state fields and methods to EditorState class

### LINKED ISSUE #125: Refactor Editor state to a new class
We currently keep a bunch of flags/fields and methods, that indicate the current state of the Editor, directly in the EditorTopComponent. To avoid this growing too large and also for the better readability, we should refactor this to a new class, called "EditorState" and keep those fields and methods there. 
Those fields include:
```
    private boolean dirty = false;
    private boolean modified = false;
    private boolean previewDisplayed = true;
    private File currentFile;
    private DbxState dbxState;
```
and all should be moved to a new class. The EditorTopComponent should expose a single method, that returns an instance of EditorState class.

### PR DESCRIPTION
**Pull request by andyzen619**
Fixes #125

### CODE REVIEW

**Comment by andyzen619** [comment]
Hi there,

This is my first pull request. I am open to any advice that you may have for me.

Thanks

File: OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorState.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (sebbrudzinski) [inline OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorState.java]: This needs proper license information. Please see other classes.

File: OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java [hunk 2]
```diff
@@ -253,11 +248,11 @@ public void setCurrentFile(File currentFile) {
     }
 
     public DbxState getDbxState() {
-        return dbxState;
+        return getEditorState().getDbxState();
     }
 
     public void setDbxState(DbxState dbxState) {
-        this.dbxState = dbxState;
+        getEditorState().setDbxState(dbxState);
```
Comment (sebbrudzinski) [inline OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java]: We should remove all of those getters/setters from this class and only have one for EditorState instance.

**Comment by sebbrudzinski** [comment]
Hi @andyzen619  thanks for the contribution. I've left the comments inline.

**Comment by sebbrudzinski** [comment]
I'll close this pull request, since we have a new one now.

**Pull request closed without merging**